inc/Engine/Preload/Fonts.php

```diff
@@ -0,0 +1,107 @@
+<?php
+
+namespace WP_Rocket\Engine\Preload;
+
+use WP_Rocket\Admin\Options_Data;
+use WP_Rocket\Event_Management\Subscriber_Interface;
+
+/**
+ * Fonts preload.
+ *
+ * @since 3.6
+ */
+class Fonts implements Subscriber_Interface {
+
+	/**
+	 * WP Rocket Options instance.
+	 *
+	 * @since 3.6
+	 *
+	 * @var Options_Data
+	 */
+	private $options;
+
+	/**
+	 * Return an array of events that this subscriber wants to listen to.
+	 *
+	 * @since  3.6
+	 *
+	 * @param Options_Data $options WP Rocket Options instance.
+	 */
+	public function __construct( Options_Data $options ) {
+		$this->options = $options;
+	}
+
+	/**
+	 * Return an array of events that this subscriber wants to listen to.
+	 *
+	 * @since  3.6
+	 *
+	 * @return array
+	 */
+	public static function get_subscribed_events() {
+		return [
+			'wp_head' => [ 'preload_fonts' ],
```

Comment:
This one is on me, verry sorry. I should have thought about this before but:

I think we should use `rocket_buffer` instead of `wp_head`. Because we:
- want the links only inserted on cached page. It's easier to handle this by using `rocket_buffer` as the hook
- want to make sure the preload links are early in the head, ideally before the inline critical CSS and the combined CSS files. I believe the loading impact wouldn't be as good if they are not first

We can use the `</title>` closing tag as the anchor to insert them.

Priority of the callback should be `20` or above.

Reply:
Whatever the preload tags for fonts are adding, it doesn't have an impact on their performance.

Here a test when the CSS files is loaded before the preload:
https://gtmetrix.com/reports/wp-rocket.me/abBweyMK

Another test with the CPCSS + CSS file loaded before the preload:
https://gtmetrix.com/reports/wp-rocket.me/nnDXH7q4

Then when the preload is added before the CSS file:
https://gtmetrix.com/reports/wp-rocket.me/VKIdXuLR

The loading time for fonts are always the same.

Why?
The CSS file is loaded, but the fonts aren't used yet.

Reply:
Thank you for the tests, I wasn't sure that was the case or not 👍